packages/terra-table/src/Table.jsx

```diff
@@ -1,68 +1,401 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames/bind';
-import TableHeader from './TableHeader';
-import TableHeaderCell from './TableHeaderCell';
-import TableRows from './TableRows';
-import TableRow from './TableRow';
-import TableCell from './TableCell';
-import SelectableTableRows from './SelectableTableRows';
-import TableSingleSelectableRows from './SingleSelectableRows';
-import TableMultiSelectableRows from './MultiSelectableRows';
-import TableSubheader from './TableSubheader';
+import ContentContainer from 'terra-content-container';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
 import styles from './Table.module.scss';
+import sectionShape from './proptypes/sectionShape';
+import headerShape from './proptypes/headerShape';
+import widthShape from './proptypes/widthShape';
+
+import Row from './subcomponents/_Row';
+import Cell from './subcomponents/_Cell';
+import Section from './subcomponents/_Section';
+import HeaderRow from './subcomponents/_HeaderRow';
+import HeaderCell from './subcomponents/_HeaderCell';
+import ChevronCell from './subcomponents/_ChevronCell';
+import CheckMarkCell from './subcomponents/_CheckMarkCell';
+import HeaderChevronCell from './subcomponents/_HeaderChevronCell';
+import HeaderCheckMarkCell from './subcomponents/_HeaderCheckMarkCell';
 
 const cx = classNames.bind(styles);
 
 const propTypes = {
   /**
-   * The children passed to the component
+   * The check mark styling to apply.
+   */
+  checkStyle: PropTypes.oneOf([
+    'none',
+    'readOnly', // TODO: address renaming this, UX ask
```

Comment:
Is this still a TOD0?

Reply:
yes - thinking we just go with `icon` as the propname (checkStyle: none / icon / toggle)
```suggestion
    'icon', 
```